pygmt/tests/test_subplot.py

```diff
@@ -23,66 +22,63 @@ def test_subplot_basic_frame():
     return fig
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_subplot_direct():
     """
     Plot map elements to subplot directly using the panel parameter.
     """
-    fig_ref, fig_test = Figure(), Figure()
-    with fig_ref.subplot(nrows=2, ncols=1, Fs="3c/3c"):
-        fig_ref.basemap(region=[0, 3, 0, 3], frame="af", panel=0)
-        fig_ref.basemap(region=[0, 3, 0, 3], frame="af", panel=1)
-    with fig_test.subplot(nrows=2, ncols=1, subsize=("3c", "3c")):
-        fig_test.basemap(region=[0, 3, 0, 3], frame="af", panel=[0, 0])
-        fig_test.basemap(region=[0, 3, 0, 3], frame="af", panel=[1, 0])
-    return fig_ref, fig_test
+    fig = Figure()
+
+    with fig.subplot(nrows=2, ncols=1, subsize=("3c", "3c")):
+        fig.basemap(region=[0, 3, 0, 3], frame="af", panel=[0, 0])
+        fig.basemap(region=[0, 3, 0, 3], frame="af", panel=[1, 0])
+    return fig
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_subplot_autolabel_margins_title():
     """
     Make subplot figure with autolabels, setting some margins and a title.
     """
-    fig_ref, fig_test = Figure(), Figure()
-    kwargs = dict(nrows=2, ncols=1, figsize=("15c", "6c"))
-
-    with fig_ref.subplot(A="a)", M="0.3c/0.1c", T="Subplot Title", **kwargs):
-        fig_ref.basemap(region=[0, 1, 2, 3], frame="WSne", c="0,0")
-        fig_ref.basemap(region=[4, 5, 6, 7], frame="WSne", c="1,0")
+    fig = Figure()
 
-    with fig_test.subplot(
-        autolabel=True, margins=["0.3c", "0.1c"], title="Subplot Title", **kwargs
+    with fig.subplot(
+        autolabel=True,
+        margins=["0.3c", "0.1c"],
+        title="Subplot Title",
+        nrows=2,
+        ncols=1,
+        figsize=("15c", "6c"),
```

Comment:
Put the required arguments (nrows/ncols/figsize) first, before the optional arguments (autolabel/margins/title).
```suggestion
        nrows=2,
        ncols=1,
        figsize=("15c", "6c"),
        autolabel=True,
        margins=["0.3c", "0.1c"],
        title="Subplot Title",
```